recipe-server/client/control/components/RecipeFormActions.js

```diff
@@ -0,0 +1,163 @@
+import React, {
+  PropTypes as pt,
+} from 'react';
+import { Link } from 'react-router';
+import cx from 'classnames';
+
+export const FormButton = ({
+    className,
+    label,
+    element = 'button',
+    type = 'button',
+    onClick,
+    display,
+    ...props,
+  }) => {
+  if (!display) {
+    return null;
+  }
+
+  // need titlecase for JSX
+  const Element = element;
+  return (
+    <Element
+      className={cx('button', className)}
+      onClick={onClick}
+      children={label}
+      type={type}
+      key={label + className}
+      {...props}
+    />
+  );
+};
+FormButton.propTypes = {
+  display: pt.bool.isRequired,
+  className: pt.string,
+  label: pt.string,
+  element: pt.any,
+  type: pt.string,
+  onClick: pt.func,
+};
+
+export default class RecipeFormActions extends React.Component {
+  static propTypes = {
+    onAction: pt.func.isRequired,
+    isUserViewingOutdated: pt.bool,
+    isPendingApproval: pt.bool,
+    isUserRequestor: pt.bool,
+    isAlreadySaved: pt.bool,
+    isFormPristine: pt.bool,
+    isCloning: pt.bool,
+    isFormDisabled: pt.bool,
+    recipeId: pt.number,
+  };
+
+  getActions({
+    isUserViewingOutdated,
+    isPendingApproval,
+    isUserRequestor,
+    isAlreadySaved,
+    isFormPristine,
+    isCloning,
+    isFormDisabled,
+    recipeId,
+  }) {
+    return [
+      // revert
```

Comment:
Reverting is just taking the currently viewed revision and saving it as a new draft. Maybe `revert` isn't the correct language?